grapher/stackedCharts/MarimekkoChart.jsdom.test.tsx

```diff
@@ -0,0 +1,340 @@
+    // Y series points should be one series in order of the data
+    const expectedYPoints = [
+        {
+            position: "medium",
+            value: 4,
+            valueOffset: 0,
+            time: 2001,
+        },
+        {
+            position: "big",
+            value: 8,
+            valueOffset: 0,
+            time: 2001,
+        },
+        {
+            position: "small",
+            value: 3,
+            valueOffset: 0,
+            time: 2001,
+        },
+    ]
+    // X series points should be in order of the data
+    const expectedXPoints = [
+        { value: 4000, entity: "medium", time: 2001 },
+        { value: 5000, entity: "big", time: 2001 },
+        { value: 1000, entity: "small", time: 2001 },
+    ]
+    expect(chart.series[0].points).toEqual(expectedYPoints)
+    expect(chart.xSeries.points).toEqual(expectedXPoints)
+    // placedItems should be in default sort order
+    expect(chart.placedItems.map(roundXPosition)).toEqual([
+        {
+            entityName: "big",
+            entityColor: undefined,
+            bars: [
+                {
+                    kind: BarShape.Bar,
+                    color: "#3C4E66",
+                    seriesName: "percentBelow2USD",
+                    yPoint: expectedYPoints[1],
+                },
+            ],
+            xPoint: expectedXPoints[1],
+            xPosition: 0,
+        },
+        {
+            entityName: "medium",
+            entityColor: undefined,
+            bars: [
+                {
+                    kind: BarShape.Bar,
+                    color: "#3C4E66",
+                    seriesName: "percentBelow2USD",
+                    yPoint: expectedYPoints[0],
+                },
+            ],
+            xPoint: expectedXPoints[0],
+            xPosition: Math.round(xAxisRange * 0.5),
+        },
+        {
+            entityName: "small",
+            entityColor: undefined,
+            bars: [
+                {
+                    kind: BarShape.Bar,
+                    color: "#3C4E66",
+                    seriesName: "percentBelow2USD",
+                    yPoint: expectedYPoints[2],
+                },
+            ],
+            xPoint: expectedXPoints[2],
+            xPosition: Math.round(xAxisRange * 0.9),
+        },
+    ])
+
+    expect(chart["xDomainCorrectionFactor"]).toEqual(1)
+})
+
+it("shows no data points at the end", () => {
+    const csv = `year,entityName,population,percentBelow2USD
+2000,medium,4000,5
+2000,big,5000,10
+2000,small,800,2
+2001,medium,4000,
+2001,big,5000,8
+2001,small,1000,3`
+    const table = new OwidTable(csv, [
+        { slug: "population", type: ColumnTypeNames.Numeric },
+        { slug: "percentBelow2USD", type: ColumnTypeNames.Numeric },
+        { slug: "year", type: ColumnTypeNames.Year },
+    ])
+
+    // TODO: why is it ySlugs and xSlug here instead of yColumnSlugs and xColumnSlug? Unify when we have config migrations?
```

Comment:
It's that validation/normalization layer that we often talk about.

`ySlugs` is the raw config property passed to the Grapher, and then we derive the "definitely correct" property in `yColumnSlugs`: https://github.com/owid/owid-grapher/blob/4e9dc0517c1c028d1bb433c829cab1b8fb3fa186/grapher/core/Grapher.tsx#L1201-L1207